File: app/src/main/java/com/notmiyouji/newsapp/java/NewsDetails/OpenNewsDetails.java
package com.notmiyouji.newsapp.java.NewsDetails;

import android.app.Activity;
import android.content.Intent;

import com.notmiyouji.newsapp.kotlin.SharedSettings.UseChromeShared;

public class OpenNewsDetails {
    private final String url;
    private final String title;
    private final String img;
    private final String source;
    private final String pubdate;
    private final Activity activity;

    public OpenNewsDetails(String url, String title, String img, String source, String pubdate, Activity activity) {
        this.url = url;
        this.title = title;
        this.img = img;
        this.source = source;
        this.pubdate = pubdate;
        this.activity = activity;
    }

    public void openNewsDetails() {
        if (new UseChromeShared(activity).getEnableChrome()) {
            //Open in chrome custom tabs
            NewsDetailsChrome chromeClient = new NewsDetailsChrome(
                    this.url,
                    this.title,
                    this.img,
                    this.url,
                    this.pubdate, activity);
            chromeClient.openNewsDetails();
        } else {
            //No chrome installed, open in webview
            Intent intent = new Intent(activity, NewsDetailWebView.class);
            intent.putExtra("url", this.url);
            intent.putExtra("title", this.title);
            intent.putExtra("img", this.img);
            intent.putExtra("source", this.source);
            intent.putExtra("pubdate", this.pubdate);
            activity.startActivity(intent);
        }
    }
}

File: app/src/main/java/com/notmiyouji/newsapp/kotlin/OpenActivity/OpenNewsDetails.kt
/*
 * Copyright By @2dgirlismywaifu (2023) .
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

package com.notmiyouji.newsapp.kotlin.OpenActivity

import android.app.Activity
import android.content.Intent
import com.notmiyouji.newsapp.java.NewsDetails.NewsDetailWebView
import com.notmiyouji.newsapp.java.NewsDetails.NewsDetailsChrome
import com.notmiyouji.newsapp.kotlin.SharedSettings.UseChromeShared

class OpenNewsDetails(
    private val url: String?,
    private val title: String?,
    private val img: String?,
    private val source: String?,
    private val pubdate: String?,
    private val activity: Activity
) {
    //function check img is null or not
    fun checkImg(): String {
        return if (img == null) {
            "https://i.imgur.com/1Z1Z1Z1.png"
        } else {
            img
        }
    }
    fun openNewsDetails() {
        when {
            UseChromeShared(activity).enableChrome -> {
                //Open in chrome custom tabs
                val chromeClient = NewsDetailsChrome(
                    url,
                    title,
                    checkImg(),
                    url,
                    pubdate, activity
                )
                chromeClient.openNewsDetails()
            }
            else -> {
                //No chrome installed, open in webview
                val intent = Intent(activity, NewsDetailWebView::class.java)
                intent.putExtra("url", url)
                intent.putExtra("title", title)
                intent.putExtra("img", checkImg())
                intent.putExtra("source", source)
                intent.putExtra("pubdate", pubdate)
                activity.startActivity(intent)
            }
        }
    }
}